### DISCUSSION TITLE
Gamepad shortcut for stop streaming

### DISCUSSION THREAD

**Discussion by mouyah** [ClassicOldSong/Apollo#118] [Q&A] [answered]
First I want to thank you for this release, really make the virtual display configuration easier for me. It's almost has all I need.

Also I want to ask if there's shortcut for gamepad to stop the streaming manually.
Usually I use it with gamepad only, and only knows L1 + R1 + Select can quit/pause the stream but unable to stop it.
So in case I run Steam Big Picture and want to stop stream without kill steam app/exe, I had to do L1+R1+Select to quit/pause and press hold A button to show menu and choose "stop stream" manually.

Sorry for bad english

**Comment by ClassicOldSong** [comment, marked as answer]
What client are you using? If you're using Android with Artemis, you can access the back menu by tap START once, then quickly hold it for a while and then release. You can end the stream directly from the quick menu.

**Reply by mouyah** [reply]
I'm using android handheld, it has integrated physical gamepad. Sorry I don't understand by START, do you mean START button on gamepad? If so, then the start button is exclusively assigned to the app (in this case steam and the games).

I mean what I did to exit in steam big pic mode now (all buttons are means to be xbox map):
1. On Big Picture main menu: back button (B) -> "Power" -> "Exit Big Picture Mode"
2. Now we still run on stream but back to windows desktop since steam exe not killed.
3. On windows desktop: L1 + R1 + Select = it back to Artemis client's menu. But the stream still running on background.
4. I have to press hold "A" button to show the menu and choose "Quit Session" to end the stream completely.

What I want is:
a. If it possible, when I choose "Exit Big Picture Mode" in step 1 above it will terminate the stream completely (like exiting the steam client behaviour). I know exiting big picture mode won't kill the exe thats why it won't trigger stop stream, I just wonder if there's a way to do that.
b. In case point "a" above is impossible, I want to use gamepad shortcut at step "3" to terminate/end the stream completely (without to do step "4")

Thank you.

**Reply by ClassicOldSong** [reply]
A is not very doable.

START is the start button on the controller. But since you're using Android handheld, you can access the back menu directly by performing a back gesture or pressing the back button.

**Reply by mouyah** [reply]
So there's no shortcut to terminate stream directly without accessing that quick menu?
There is no back button in my handheld, and I want to avoid using touchscreen/gesture as much as possible.
Is it possible to make "terminate stream session" shortcut feature in the future (hopefully)?

Also I checked on moonlight wiki, it said: Ctrl+Alt+Shift+Q - Quit the streaming session (leaving the game running on the host PC).
I tried it, and it seems this keyboard shortcut just did the same, only going back to moonlight/artemis screen without quit the stream session completely.

**Reply by ClassicOldSong** [reply]
There isn't.

Too much shortcut will overwhelm users and increase conflicts if more and more layers of software you use to tweak your controller input. The reason I decided to use the tap and hold START pattern is:

1. Most times clicking START won't do disastrous things to your game play, it'll only pause or pop up in game menu.
2. If you start holding START within the tap hold pattern timing threshold, no start button event will be sent to your host when you perform hold, so it will not interrupt the pause state triggered by the previous tap action.
3. Each tool/layer wants to take some shortcuts to do something, but a button sequence isn't widely used so it's most likely won't conflict with any other shortcuts on the host, and leaving more room for customization.

Also you can terminate a stream directly from Apollo's tray menu. I added this functionality in particular for the convenience of quitting apps.

**Comment by mouyah** [comment]
ok thank you for the explanation, I'm going to use that double press and hold START button.
